sonic-xcvrd/xcvrd/xcvrd_utilities/media_settings_parser.py

```diff
@@ -231,60 +235,133 @@ def get_media_settings(key, media_dict):
     # 1,2,3,4,5
     # 1-4,9-12
 
-    if GLOBAL_MEDIA_SETTINGS_KEY in g_dict:
-        for keys in g_dict[GLOBAL_MEDIA_SETTINGS_KEY]:
-            if COMMA_SEPARATOR in keys:
-                port_list = keys.split(COMMA_SEPARATOR)
-                for port in port_list:
-                    if RANGE_SEPARATOR in port:
-                        if xcvrd.check_port_in_range(port, physical_port):
-                            media_dict = g_dict[GLOBAL_MEDIA_SETTINGS_KEY][keys]
+    if gearbox_side:
+        if GEARBOX_GLOBAL_MEDIA_SETTINGS_KEY in g_dict:
```

Comment:
@j-mazurek can you write a separate function `get_gearbox_media_settings()` do the necessary refactor?